File: Code_CS/C17_Caching/SimpleUserControl.ascx.cs
using System;
using System.Web.UI;

public partial class SimpleUserControl : UserControl
{
    public string UserName
    {
        get { return lblUserName.Text; }
        set { lblUserName.Text = value; }
    }

   protected void Page_Load(object sender, EventArgs e)
   {
      lblTime.Text = String.Format("Control time is {0}",
         DateTime.Now.ToLongTimeString());
   }

   protected void btnChange_Click(object sender, EventArgs e)
   {
       lblUserName.Text = "Janey";
   }
}

File: Code_VB/C17_Caching/SimpleUserControl.ascx.vb

Partial Class SimpleUserControl
    Inherits System.Web.UI.UserControl

    Public Property UserName() As String
        Get
            Return lblUserName.Text
        End Get
        Set(ByVal value As String)
            lblUserName.Text = value
        End Set
    End Property

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        lblTime.Text = String.Format("Control time is {0}", DateTime.Now.ToLongTimeString())

    End Sub

    Protected Sub btnChange_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnChange.Click

        lblUserName.Text = "Janey"

    End Sub

End Class
